src/bricks/transformers/brickFactory.ts

```
@@ -211,7 +213,7 @@ class UserDefinedBrick extends BrickABC {
     const awareness = await Promise.all(
       pipeline.map(async (blockConfig) => {
         const resolvedBlock = await this.registry.lookup(blockConfig.id);
-        return resolvedBlock.isPageStateAware();
+        return resolvedBlock?.isPageStateAware?.();
```

This is changing the behavior. Instead of throwing an error it's making it return false

If we don't want it to throw an error, I'd recommend we instead have it fall back to `true`. `true` is a safer default (it indicates it "might" read from page state)

---

Why does this only come up in this method and not isPure, isRootAware, etc?: https://github.com/pixiebrix/pixiebrix-extension/pull/8456/files#diff-fac01820dc5aba759155a71b60958ca6c0dd781d05763dac57041c40ec0b63b3R190

I'd recommend we keep the behavior consistent. I'd keep it consistent as a DoesNotExistError

---

Looks like we changed `isPure` and `isRootAware` to be non-optional 6 months ago as part of earlier Strict Null Checks work. I can do the same here since we'd rather throw an error